enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py

```diff
@@ -0,0 +1,209 @@
+                        except Subsidy.DoesNotExist:
+                            logger.warning(
+                                f"{DRY_RUN_LOG_PREFIX}Unable to determine enterprise customer uuid for transaction: "
+                                f"{related_transaction}"
+                            )
+                            continue
+
+                        content_metadata = enterprise_catalog_api_client.get_content_metadata_for_customer(
+                            customer_uuid,
+                            enrollment_course_run_key,
+                        )
+                        fetched_content_metadata[enrollment_course_run_key] = content_metadata
+                    else:
+                        content_metadata = fetched_content_metadata.get(enrollment_course_run_key)
+
+                    if self.determine_refundability_of_unenrollment(content_metadata, enterprise_course_enrollment):
+                        logger.info(
+                            f"{DRY_RUN_LOG_PREFIX}Course run: {enrollment_course_run_key} is refundable for enterprise"
+                            f"customer user: {enterprise_course_enrollment.get('enterprise_customer_user')}. Writing "
+                            "Reversal record."
+                        )
+
+                        if not dry_run:
+                            # Do we need to write any additional metadata to the Reversal?
+                            reverse_full_transaction(
+                                transaction=related_transaction,
+                                idempotency_key=idempotency_key,
+                            )
+                        else:
+                            logger.info(
+                                f"{DRY_RUN_LOG_PREFIX}Would have written Reversal record for enterprise fulfillment: "
+                                f"{fulfillment_uuid}. Transaction: {related_transaction}."
+                            )
+                        reversals_processed += 1
+                    else:
+                        logger.info(
+                            f"{DRY_RUN_LOG_PREFIX}Unenrollment from course: {enrollment_course_run_key} by user: "
+                            f"{enterprise_course_enrollment.get('enterprise_customer_user')} is not refundable."
+                        )
+                else:
+                    logger.info(
+                        f"{DRY_RUN_LOG_PREFIX}Found unenrolled enterprise fulfillment: {fulfillment_uuid} related to "
+                        f"an externally referenced transaction: {related_transaction.external_reference.first()}. "
+                        "Skipping."
+                    )
+            else:
+                logger.info(
+                    f"{DRY_RUN_LOG_PREFIX}Found existing Reversal: {existing_reversal} for enterprise fulfillment: "
+                    f"{fulfillment_uuid}. Skipping Reversal creation for Transaction: {related_transaction}."
+                )
```

Comment:
There is a lot of nesting to this logic.  Is there any low-hanging-fruit way to simplify or help make it more readable?